core/metrics/src/main/scala/org/typelevel/otel4s/metrics/meta/InstrumentMeta.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.metrics.meta
+
+import cats.Applicative
+import cats.Monad
+import cats.~>
+import org.typelevel.otel4s.KindTransformer
+
+sealed trait InstrumentMeta[F[_]] extends org.typelevel.otel4s.meta.InstrumentMeta.Dynamic.Unsealed[F] {
+
+  /** Indicates whether instrumentation is enabled or not.
+    */
+  def isEnabled: F[Boolean]
+
+  /** A no-op effect.
+    */
+  def unit: F[Unit]
+
+  /** Runs `f` only when the instrument is enabled. A shortcut for `Monad[F].ifM(isEnabled)(f, unit)`.
+    */
+  def whenEnabled(f: => F[Unit]): F[Unit]
+
+  /** Modify the context `F` using the transformation `f`. */
+  @deprecated("use `liftTo` instead", since = "otel4s 0.14.0")
+  override def mapK[G[_]: Monad](f: F ~> G): InstrumentMeta[G] =
+    new InstrumentMeta.MappedK(this)(f)
```

Comment:
We can remove `mapK` eventually because it's not used in the metrics module.

Reply:
once we can depend on cats-mtl:1.6.0, I plan to deprecate the `mapK[G[_]](F ~> G)` methods as no longer supported for the future, and point to `liftTo[G[_]](implicit LiftValue[F, G])` instead